src/control/IntersectionReferenceSystem.ts

```diff
@@ -277,8 +305,8 @@ export class IntersectionReferenceSystem {
     const refEnd = new Vector2(this.interpolators.trajectory.getPointAt(1.0));
     const endVec = new Vector2(this.endVector);
     const postStep = extensionEnd / postSteps;
-    for (let i = 1; i < postSteps; i++) {
-      const f = i * postStep;
+    for (let i = 0; i < postSteps; i++) {
```

Comment:
What would you call map on in this case?

Reply:
I guess you @nilsml would like to introduce some form of range(from, to) to get rid of the loop. I will not introduce it now just for this case, but let's consider if it's worth doing that down the road.

Reply:
I like the idea of range(0, n) like in c#, however I have not found an elegant solution in js without allocating an array of n element. That may have an impact on performance so care need to be taken. If there is a better solution please let me know.

Reply:
Yes, you'd have to go through an array @HavardNJ, so performance could become an issue if it gets called a lot.
`[...Array(postSteps).keys()].map(...` or similar would work.

Not a very important improvement though.

Reply:
So Array() create a new array, then keys() create an iterator and the spread create a second array by iterating over the keys and then you can start the iteration you are interested in. I would think that Array(steps).fill().map((_, i) => ... or [...Array(steps)].map((_, i) => ... would be more performant though I havn't tested. Not convinced this is more readable/understandable than a simple for loop for anyone who have not come across this before, though it could be hidden behind a range function/class.

Reply:
Agree about readability. The syntax we are missing in js is `[0..n].map(`. In this case I would have hidden this logic in a function somewhere and used it here. The reason why I promote code like this is because I think it is better to show intention than how we do things. Functional over imperative programming. There are many ways to achieve this. Getting rid of for-loops is usually a good start.
Also map produces a single result. We don't have to mutate a variable for each iteration.